src/hotspot/share/opto/superword.cpp

```diff
@@ -2346,6 +2348,204 @@ void SuperWord::verify_packs() {
+      }
+    }
+
+    int max_pid_packset = _max_pid;
+
+    // Map nodes not in packset
+    for (int i = 0; i < _block.length(); i++) {
+      Node* n = _block.at(i);
+      if (n->is_Phi() || n->is_CFG()) {
+        continue; // ignore control flow
+      }
+      int pid = get_pid_or_zero(n);
+      if (pid == 0) {
+        pid = new_pid();
+        set_pid(n, pid);
+      }
+    }
+
+    // Map edges for packset nodes
+    VectorSet set;
+    for (int i = 0; i < _packset.length(); i++) {
+      Node_List* p = _packset.at(i);
+      set.clear();
+      for (uint k = 0; k < p->size(); k++) {
+        Node* n = p->at(k);
+        int pid = get_pid(n);
+        for (DepPreds preds(n, _dg); !preds.done(); preds.next()) {
+          Node* pred = preds.current();
+          int pred_pid = get_pid_or_zero(pred);
+          if (pred_pid == pid && n->is_reduction()) {
+            continue; // reduction -> self-cycle is not a cyclic dependency
+          }
+          // Only add edges once, and only for mapped nodes (in _block)
+          if (pred_pid > 0 && !set.test_set(pred_pid)) {
+            incnt_set(pid, incnt(pid) + 1); // increment
+            out(pred_pid).push(pid);
+          }
+        }
+      }
+    }
+
+    // Map edges for nodes not in packset
+    for (int i = 0; i < _block.length(); i++) {
+      Node* n = _block.at(i);
+      int pid = get_pid_or_zero(n);
+      if (pid == 0 || pid <= max_pid_packset) {
+        continue; // Only scalar-nodes
+      }
+      for (DepPreds preds(n, _dg); !preds.done(); preds.next()) {
+        Node* pred = preds.current();
+        int pred_pid = get_pid_or_zero(pred);
+        // Only add edges for mapped nodes (in _block)
+        if (pred_pid > 0) {
+          incnt_set(pid, incnt(pid) + 1); // increment
+          out(pred_pid).push(pid);
+        }
+      }
+    }
+  }
+  // Schedule the graph to worklist. Returns true iff all nodes were scheduled.
+  // This implies that we return true iff the PacksetGraph is acyclic.
+  // We schedule with topological sort: schedule any node that has zero incnt.
+  // Then remove that node, which decrements the incnt of all its uses (outputs).
+  bool schedule() {
+    GrowableArray<int> worklist;
+    // Directly schedule all nodes without precedence
+    for (int pid = 1; pid <= _max_pid; pid++) {
+      if (incnt(pid) == 0) {
+        worklist.push(pid);
+      }
+    }
+    // Continue scheduling via topological sort
+    for (int i = 0; i < worklist.length(); i++) {
+      int pid = worklist.at(i);
+      for (int j = 0; j < out(pid).length(); j++){
+        int pid_use = out(pid).at(j);
+        int incnt_use = incnt(pid_use) - 1;
+        incnt_set(pid_use, incnt_use);
+        // Did use lose its last input?
+        if (incnt_use == 0) {
+          worklist.push(pid_use);
+        }
+      }
+    }
+    // Was every pid scheduled?
+    return worklist.length() == _max_pid;
+  }
+  void print(bool print_nodes, bool print_zero_incnt) {
```

Comment:
I assume you need this parameters for debugging when you can change their values.

Reply:
Exactly. This helps debugging the scheduling. I added a comment to explain the arguments.